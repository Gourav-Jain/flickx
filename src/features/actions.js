/**
 * Action Types
 */
export const SEARCH_MOVIES_REQS = "SEARCH_MOVIES_REQS";
export const SEARCH_MOVIES_RESP = "SEARCH_MOVIES_RESP";
export const SEARCH_MOVIES_FAIL = "SEARCH_MOVIES_FAIL";

export const ADD_TO_FAVOURITE = "ADD_TO_FAVOURITE";
export const ADD_TO_WATCH_LATER = "ADD_TO_WATCH_LATER";
export const REMOVE_FROM_FAVOURITE = "REMOVE_FROM_FAVOURITE";
export const REMOVE_FROM_WATCH_LATER = "REMOVE_FROM_WATCH_LATER";

/**
 * Action Creator
 */
export const setSearchMoviesRequest = payload => ({ type: SEARCH_MOVIES_REQS, payload });
export const setSearchMoviesResponse = payload => ({ type: SEARCH_MOVIES_RESP, payload });
export const setSearchMoviesFailer = payload => ({ type: SEARCH_MOVIES_FAIL, payload });

export const setAddToFavourites = payload => ({ type: ADD_TO_FAVOURITE, payload });
export const setAddToWatchLater = payload => ({ type: ADD_TO_WATCH_LATER, payload });
export const setRemoveFromFavourite = payload => ({ type: REMOVE_FROM_FAVOURITE, payload });
export const setRemoveFromWatchLater = payload => ({ type: REMOVE_FROM_WATCH_LATER, payload });