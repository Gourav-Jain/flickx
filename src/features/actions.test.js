import * as actions from "./actions";

describe("@actions and @creators", () => {
    describe("actions", () => {
        const searchActions = [actions.SEARCH_MOVIES_REQS,
        actions.SEARCH_MOVIES_RESP,
        actions.SEARCH_MOVIES_FAIL,
        actions.ADD_TO_FAVOURITE,
        actions.ADD_TO_WATCH_LATER,
        actions.REMOVE_FROM_FAVOURITE,
        actions.REMOVE_FROM_WATCH_LATER];

        test.each(searchActions)("should return %s action", actionType =>
            expect(actions[actionType]).toMatchSnapshot()
        );
    });

    describe("creators", () => {
        test.each`
      creatorName       | mockData
      ${"setSearchMoviesRequest"} | ${"mock-data"}
      ${"setSearchMoviesResponse"} | ${"mock-data"}
      ${"setSearchMoviesFailer"} | ${"mock-data"}
      ${"setAddToFavourites"} | ${"mock-data"}
      ${"setAddToWatchLater"} | ${"mock-data"}
      ${"setRemoveFromFavourite"} | ${"mock-data"}
      ${"setRemoveFromWatchLater"} | ${"mock-data"}
    `("$creatorName returns correctly", ({ creatorName, mockData }) => {
            expect(actions[creatorName](mockData)).toMatchSnapshot();
        });
    });
});