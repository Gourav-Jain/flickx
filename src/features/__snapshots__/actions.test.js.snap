// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@actions and @creators actions should return ADD_TO_FAVOURITE action 1`] = `"ADD_TO_FAVOURITE"`;

exports[`@actions and @creators actions should return ADD_TO_WATCH_LATER action 1`] = `"ADD_TO_WATCH_LATER"`;

exports[`@actions and @creators actions should return REMOVE_FROM_FAVOURITE action 1`] = `"REMOVE_FROM_FAVOURITE"`;

exports[`@actions and @creators actions should return REMOVE_FROM_WATCH_LATER action 1`] = `"REMOVE_FROM_WATCH_LATER"`;

exports[`@actions and @creators actions should return SEARCH_MOVIES_FAIL action 1`] = `"SEARCH_MOVIES_FAIL"`;

exports[`@actions and @creators actions should return SEARCH_MOVIES_REQS action 1`] = `"SEARCH_MOVIES_REQS"`;

exports[`@actions and @creators actions should return SEARCH_MOVIES_RESP action 1`] = `"SEARCH_MOVIES_RESP"`;

exports[`@actions and @creators creators setAddToFavourites returns correctly 1`] = `
Object {
  "payload": "mock-data",
  "type": "ADD_TO_FAVOURITE",
}
`;

exports[`@actions and @creators creators setAddToWatchLater returns correctly 1`] = `
Object {
  "payload": "mock-data",
  "type": "ADD_TO_WATCH_LATER",
}
`;

exports[`@actions and @creators creators setRemoveFromFavourite returns correctly 1`] = `
Object {
  "payload": "mock-data",
  "type": "REMOVE_FROM_FAVOURITE",
}
`;

exports[`@actions and @creators creators setRemoveFromWatchLater returns correctly 1`] = `
Object {
  "payload": "mock-data",
  "type": "REMOVE_FROM_WATCH_LATER",
}
`;

exports[`@actions and @creators creators setSearchMoviesFailer returns correctly 1`] = `
Object {
  "payload": "mock-data",
  "type": "SEARCH_MOVIES_FAIL",
}
`;

exports[`@actions and @creators creators setSearchMoviesRequest returns correctly 1`] = `
Object {
  "payload": "mock-data",
  "type": "SEARCH_MOVIES_REQS",
}
`;

exports[`@actions and @creators creators setSearchMoviesResponse returns correctly 1`] = `
Object {
  "payload": "mock-data",
  "type": "SEARCH_MOVIES_RESP",
}
`;
